test/Language/PureScript/Scheme/MakeSpec.hs

```diff
@@ -1,29 +1,37 @@
 module Language.PureScript.Scheme.MakeSpec (spec) where
 
 import qualified Data.Text.IO                    as T.IO
-import           Test.Hspec                      (Spec, it, shouldBe)
+import           Test.Hspec                      (Spec, it, shouldBe, beforeAll_)
 import           Language.PureScript.Scheme.Make (compile)
+import qualified Turtle
 
-basePath :: String
-basePath = "test/resources"
-  
 modules :: [String]
 modules = [ "PureScmTest.Literals"
           , "PureScmTest.Fib"
           , "PureScmTest.FunctionMultipleMatch" ]
 
 coreFnPath :: String -> FilePath
-coreFnPath module_ = basePath <> "/" <> module_ <> ".corefn.json"
+coreFnPath module_ = "test/test-project/output/" <> module_ <> "/corefn.json"
 
 schemePath :: String -> FilePath
-schemePath module_ = basePath <> "/" <> module_ <> ".sls"
+schemePath module_ = "test/fixtures/" <> module_ <> ".sls"
 
 testModule :: String -> Spec
 testModule module_ = do
   it ("Can compile " <> module_) $ do
     compilationResult <- compile $ coreFnPath module_
     expectedResult <- T.IO.readFile $ schemePath module_
     compilationResult `shouldBe` expectedResult
-  
+
+setup :: IO ()
+setup = do
+  oldDir <- Turtle.pwd
```

Comment:
What about something like (maybe to have in a separate file?)

```
inDir :: Turtle.MonadIO m => Turtle.FilePath -> m b -> m b
inDir directory do_ = do
  oldDir <- Turtle.pwd
  Turtle.cd directory
  res <- do_
  Turtle.cd oldDir
  return res

runInDir
  :: Turtle.MonadIO m
  => Turtle.FilePath
  -> Turtle.Text
  -> m (Turtle.ExitCode, Turtle.Text, Turtle.Text)
runInDir directory cmd = inDir directory $ Turtle.shellStrictWithErr cmd mempty
```

and then

```
setup :: IO ()
setup = do
  (exitCode, _out, err) <- runInDir "test/test-project" "spago build -u '-g corefn'"
  case exitCode of
    Turtle.ExitSuccess -> pure ()
    Turtle.ExitFailure _ -> error $ "spago failed to run. error: " <> show err
```

It would also be nice to have the constants factored out.